implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteTagNameMapper.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.graphite;
+
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.config.NamingConvention;
+import io.micrometer.core.instrument.util.HierarchicalNameMapper;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Tag based {@link HierarchicalNameMapper} for Graphite.
+ *
+ * @author Jon Schneider
+ * @author Johnny Lim
+ * @author Andrew Fitzgerald
```

Comment:
Copy-Paste-Modify from the `GraphiteHierarchicalNameMapper` (same story for the test)

Reply:
No issue other than just please update the year on the license header at top.

Reply:
5b5a6ae78ca4e56213df8e222b1113adeb26aa3c